### DISCUSSION TITLE
Blade's {{ }} echo statements should automatically handle enum types

### DISCUSSION THREAD

**Discussion by yob-yob** [laravel/framework#39895] [Ideas]
Since Php 8.1 has already been released and Enum types is already a thing...

I think it's best that Blade's `{{ }}` echo statements should automatically handle enum types...

implementing a `__toString()` method in an enum throws a  `Enum may not include __toString `... so using enums in `{{}}` throws  a `Argument #1 ($string) must be of type string, EnumType given in /vendor/laravel/framework/src/Illuminate/Support/helpers.php:118`

Laravel already supports `Casting` model attributes to `Enums`, it would also be much easier if `{{ }}` handles `$model->enum->value` automatically...

**Comment by ipontt** [comment]
If I recall correctly, `{{ }}` uses the `e()` function defined in the `Illuminate\Support\helpers.php` file. The error seems to be caused by the underlying call to `htmlspecialchars`.

```php
enum Letters
{
    case A;
    case B;
    case C;
}

$letter = Letter::A;

e($letter); // throws the error Argument #1 ($string) must be of type string, Letter given
e($letter->name); // echoes the string 'A'.
```
The `e()` function doesn't check if what you pass into it is an `Enum` or not. It checks if it's an instance of `Illuminate\Contracts\Support\DeferringDisplayableValue` or `Illuminate\Contracts\Support\Htmlable`. You could pick one of these 2 interfaces and implement them in your `Enum`.

```php 
// Solution 1
enum Letters implements \Illuminate\Contracts\Support\DeferringDisplayableValue
{
    case A;
    case B;
    case C;

    public function resolveDisplayableValue()
    {
        return $this->name;
    }
}

$letter = Letter::A;

e($letter); // echoes the string 'A'.
```
.

```php 
// Solution 2
enum Letters implements \Illuminate\Contracts\Support\Htmlable
{
    case A;
    case B;
    case C;

    public function toHtml()
    {
        return $this->name;
    }
}

$letter = Letter::A;

e($letter); // echoes the string 'A'.
```

**Reply by luckydonald** [reply]
While that is a workaround, I still think this should be part of the Blade renderer.
For example right now you can put non-`string`s into there, like `integer`s. They work too. 

Besides it's a maintenance burden, adding that to 100+ Enums is cumbersome, too.

**Comment by henzeb** [comment]
My package currently supports what you propose https://github.com/henzeb/enumhancer/blob/main/docs/blade.md
